diff -urNp -x '*.orig' postgresql-9.3.25.org/configure.in postgresql-9.3.25/configure.in
--- postgresql-9.3.25.org/configure.in	2018-11-05 22:53:28.000000000 +0100
+++ postgresql-9.3.25/configure.in	2021-11-16 21:00:35.767800068 +0100
@@ -19,10 +19,10 @@ m4_pattern_forbid(^PGAC_)dnl to catch un
 
 AC_INIT([PostgreSQL], [9.3.25], [pgsql-bugs@postgresql.org])
 
-m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.63], [], [m4_fatal([Autoconf version 2.63 is required.
-Untested combinations of 'autoconf' and PostgreSQL versions are not
-recommended.  You can remove the check from 'configure.in' but it is then
-your responsibility whether the result works or not.])])
+dnl m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.63], [], [m4_fatal([Autoconf version 2.63 is required.
+dnl Untested combinations of 'autoconf' and PostgreSQL versions are not
+dnl recommended.  You can remove the check from 'configure.in' but it is then
+dnl your responsibility whether the result works or not.])])
 AC_COPYRIGHT([Copyright (c) 1996-2013, PostgreSQL Global Development Group])
 AC_CONFIG_SRCDIR([src/backend/access/common/heaptuple.c])
 AC_CONFIG_AUX_DIR(config)
