--- postgresql-7.2/src/backend/libpq/pg_hba.conf.sample.sec	Fri Feb  8 14:52:17 2002
+++ postgresql-7.2/src/backend/libpq/pg_hba.conf.sample	Fri Feb  8 14:53:12 2002
@@ -241,5 +241,12 @@
 # 
 # TYPE     DATABASE    IP_ADDRESS    MASK               AUTH_TYPE  AUTH_ARGUMENT
 
-local      all                                          trust
-host       all         127.0.0.1     255.255.255.255    trust
+#local      all                                          trust
+#host       all         127.0.0.1     255.255.255.255    trust
+
+# Using sockets credentials for improved security. Not available everywhere,
+# but works on Linux, *BSD (and probably some others)
+
+local	all	ident	sameuser
+
+
