--- postgresql-7.4beta1/src/backend/utils/misc/postgresql.conf.sample	2003-08-08 22:41:16.000000000 +0200
+++ postgresql-7.4beta1/src/backend/utils/misc/postgresql.conf.sample.new	2003-08-08 22:40:08.000000000 +0200
@@ -137,9 +137,9 @@
 
 # - Syslog -
 
-#syslog = 0			# range 0-2; 0=stdout; 1=both; 2=syslog
-#syslog_facility = 'LOCAL0'
-#syslog_ident = 'postgres'
+syslog = 2			# range 0-2; 0=stdout; 1=both; 2=syslog
+syslog_facility = 'LOCAL0'
+syslog_ident = 'postgres'
 
 # - When to Log -
 
@@ -170,9 +170,9 @@
 #debug_print_rewritten = false
 #debug_print_plan = false
 #debug_pretty_print = false
-#log_connections = false
+log_connections = true
 #log_duration = false
-#log_pid = false
+log_pid = true 
 #log_statement = false
 #log_timestamp = false
 #log_hostname = false
