diff -Nru postgresql-7.0.2/src/interfaces/python/mkdefines postgresql-7.0.2.new/src/interfaces/python/mkdefines
--- postgresql-7.0.2/src/interfaces/python/mkdefines	Mon May 17 08:15:27 1999
+++ postgresql-7.0.2.new/src/interfaces/python/mkdefines	Thu Jul  6 02:34:09 2000
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python
 
 import string
 
diff -Nru postgresql-7.0.2/src/interfaces/python/tutorial/advanced.py postgresql-7.0.2.new/src/interfaces/python/tutorial/advanced.py
--- postgresql-7.0.2/src/interfaces/python/tutorial/advanced.py	Mon May 17 08:06:28 1999
+++ postgresql-7.0.2.new/src/interfaces/python/tutorial/advanced.py	Thu Jul  6 02:32:57 2000
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python
 # advanced.py - demo of advanced features of PostGres. Some may not be ANSI.
 # inspired from the Postgres tutorial 
 # adapted to Python 1995 by Pascal Andre
diff -Nru postgresql-7.0.2/src/interfaces/python/tutorial/basics.py postgresql-7.0.2.new/src/interfaces/python/tutorial/basics.py
--- postgresql-7.0.2/src/interfaces/python/tutorial/basics.py	Mon May 17 08:06:30 1999
+++ postgresql-7.0.2.new/src/interfaces/python/tutorial/basics.py	Thu Jul  6 02:32:50 2000
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python
 # basics.py - basic SQL commands tutorial
 # inspired from the Postgres95 tutorial 
 # adapted to Python 1995 by Pascal ANDRE
diff -Nru postgresql-7.0.2/src/interfaces/python/tutorial/pgtools.py postgresql-7.0.2.new/src/interfaces/python/tutorial/pgtools.py
--- postgresql-7.0.2/src/interfaces/python/tutorial/pgtools.py	Mon May 17 08:06:34 1999
+++ postgresql-7.0.2.new/src/interfaces/python/tutorial/pgtools.py	Thu Jul  6 02:32:46 2000
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python
 # pgtools.py - valuable functions for PostGreSQL tutorial
 # written 1995 by Pascal ANDRE
 
