diff -Nru postgresql-7.0.2/doc/Makefile postgresql-7.0.2.new/doc/Makefile
--- postgresql-7.0.2/doc/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/doc/Makefile	Sat Jul 22 06:29:39 2000
@@ -57,8 +57,8 @@
 	$(MAKE) clean
 
 man::
-	-@if test ! -d $(POSTMANDIR) ; then mkdir $(POSTMANDIR) ; fi
-	$(GZCAT) man.tar.gz | (cd $(POSTMANDIR) ; $(TAR) xf - )
+	mkdir -p $(DESTDIR)$(POSTMANDIR)
+	$(GZCAT) man.tar.gz | (cd $(DESTDIR)$(POSTMANDIR) ; $(TAR) xf - )
 
 #
 # Generic production rules
@@ -73,7 +73,6 @@
 #  to replace symlinks - thomas 1998-03-01
 
 %:	%.tar.gz
-	rm -rf ./$@ $(PGDOCS)/$*
-	if test ! -d $(PGDOCS)/$* ; then mkdir $(PGDOCS)/$* ; fi
-	$(GZCAT) $< | (cd $(PGDOCS)/$* ; $(TAR) xf - )
+	mkdir -p $(DESTDIR)$(PGDOCS)/$*
+	$(GZCAT) $< | (cd $(DESTDIR)$(PGDOCS)/$* ; $(TAR) xf - )
 #	touch ./$*
diff -Nru postgresql-7.0.2/src/Makefile.global.in postgresql-7.0.2.new/src/Makefile.global.in
--- postgresql-7.0.2/src/Makefile.global.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/Makefile.global.in	Mon Jul 24 11:01:33 2000
@@ -66,13 +66,14 @@
 
 # Where the postgres executables live (changeable by just putting them
 # somewhere else and putting that directory in your shell PATH)
-BINDIR= $(POSTGRESDIR)/bin
+BINDIR= @bindir@
 
 # Where libpq.a gets installed.  You must put it where your loader will
 # look for it if you wish to use the -lpq convention.  Otherwise you
 # can just put the absolute pathname to the library at the end of your
 # command line.
-LIBDIR= $(POSTGRESDIR)/lib
+LIBDIR= @libdir@
+sysconfdir= @sysconfdir@
 
 # Where the database templates are stored
 #
@@ -84,13 +85,13 @@
 IPCRM=@ipcrm@
 
 # Where the man pages (suitable for use with "man") get installed.
-POSTMANDIR= $(POSTGRESDIR)/man
+POSTMANDIR= @mandir@
 
 # Where the formatted documents (e.g., the reference manual) get installed.
 POSTDOCDIR= $(POSTGRESDIR)/doc
 
 # Where the header files necessary to build frontend programs get installed.
-HEADERDIR= $(POSTGRESDIR)/include
+HEADERDIR= @includedir@/pgsql
 
 # Where the ODBC installation-wide configuration file gets installed.
 # This needs to be a client-side readable area, so can't go in $PGDATA.
diff -Nru postgresql-7.0.2/src/Makefile.shlib postgresql-7.0.2.new/src/Makefile.shlib
--- postgresql-7.0.2/src/Makefile.shlib	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/Makefile.shlib	Thu Jul  6 10:01:02 2000
@@ -269,18 +269,20 @@
 .PHONY: all install-lib install-shlib
 
 install-lib: lib$(NAME).a
-	$(INSTALL) $(INSTL_LIB_OPTS) lib$(NAME).a $(LIBDIR)/lib$(NAME).a
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)
+	$(INSTALL) $(INSTL_LIB_OPTS) lib$(NAME).a $(DESTDIR)$(LIBDIR)/lib$(NAME).a
 
 install-shlib: $(shlib)
-	$(INSTALL) $(INSTL_SHLIB_OPTS) $(shlib) $(LIBDIR)/$(shlib)
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)
+	$(INSTALL) $(INSTL_SHLIB_OPTS) $(shlib) $(DESTDIR)$(LIBDIR)
 ifneq ($(PORTNAME), win)
 	if [ "$(shlib)" != "lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION)" ]; then \
-		cd $(LIBDIR); \
+		cd $(DESTDIR)$(LIBDIR); \
 		rm -f lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION); \
 		$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION); \
 	fi
 	if [ "$(shlib)" != "lib$(NAME)$(DLSUFFIX)" ]; then \
-		cd $(LIBDIR); \
+		cd $(DESTDIR)$(LIBDIR); \
 		rm -f lib$(NAME)$(DLSUFFIX); \
 		$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX); \
 	fi
diff -Nru postgresql-7.0.2/src/backend/Makefile postgresql-7.0.2.new/src/backend/Makefile
--- postgresql-7.0.2/src/backend/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/backend/Makefile	Sat Jul 22 07:13:08 2000
@@ -167,118 +167,91 @@
 
 .PHONY: install install-bin install-templates install-headers
 
-install: $(LIBDIR) $(BINDIR) $(HEADERDIR) $(TEMPLATEDIR) \
-         postgres $(POSTGRES_IMP) \
+install: postgres $(POSTGRES_IMP) \
          install-bin install-templates install-headers
 
 install-bin: $(BINDIR) postgres$(X) $(POSTGRES_IMP)
-	$(INSTALL) $(INSTL_EXE_OPTS) postgres$(X) $(BINDIR)/postgres$(X)
-	@rm -f $(BINDIR)/postmaster
-	ln -s postgres$(X) $(BINDIR)/postmaster
+	$(INSTALL) -d $(DESTDIR)$(BINDIR) $(DESTDIR)$(LIBDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) postgres$(X) $(DESTDIR)$(BINDIR)/postgres$(X)
+	ln -s postgres$(X) $(DESTDIR)$(BINDIR)/postmaster
 ifeq ($(MAKE_EXPORTS), true)
-	$(INSTALL) $(INSTLOPTS) $(POSTGRES_IMP) $(LIBDIR)/$(POSTGRES_IMP)
+	$(INSTALL) $(INSTLOPTS) $(POSTGRES_IMP) $(DESTDIR)$(LIBDIR)/$(POSTGRES_IMP)
 endif
 ifeq ($(PORTNAME), win)
 ifeq ($(MAKE_DLL), true)
-#	$(INSTALL) $(INSTLOPTS) postgres.dll $(BINDIR)/postgres.dll
-	$(INSTALL) $(INSTLOPTS) libpostgres.a $(LIBDIR)/libpostgres.a
+#	$(INSTALL) $(INSTLOPTS) postgres.dll $(DESTDIR)$(BINDIR)/postgres.dll
+	$(INSTALL) $(INSTLOPTS) libpostgres.a $(DESTDIR)$(LIBDIR)/libpostgres.a
 endif
 endif
 
-install-templates: $(TEMPLATEDIR) \
+install-templates: \
          global1.bki.source local1_template1.bki.source \
          global1.description local1_template1.description \
          libpq/pg_hba.conf.sample libpq/pg_ident.conf.sample \
 	 optimizer/geqo/pg_geqo.sample \
          pg_options.sample
+	$(INSTALL) -d $(DESTDIR)$(TEMPLATEDIR)
 	$(INSTALL) $(INSTLOPTS) global1.bki.source \
-	  $(TEMPLATEDIR)/global1.bki.source
+	  $(DESTDIR)$(TEMPLATEDIR)/global1.bki.source
 	$(INSTALL) $(INSTLOPTS) global1.description \
-	  $(TEMPLATEDIR)/global1.description
+	  $(DESTDIR)$(TEMPLATEDIR)/global1.description
 	$(INSTALL) $(INSTLOPTS) local1_template1.bki.source \
-	  $(TEMPLATEDIR)/local1_template1.bki.source
+	  $(DESTDIR)$(TEMPLATEDIR)/local1_template1.bki.source
 	$(INSTALL) $(INSTLOPTS) local1_template1.description \
-	  $(TEMPLATEDIR)/local1_template1.description
+	  $(DESTDIR)$(TEMPLATEDIR)/local1_template1.description
 	$(INSTALL) $(INSTLOPTS) libpq/pg_hba.conf.sample \
-	  $(TEMPLATEDIR)/pg_hba.conf.sample
+	  $(DESTDIR)$(TEMPLATEDIR)/pg_hba.conf.sample
 	$(INSTALL) $(INSTLOPTS) libpq/pg_ident.conf.sample \
-	  $(TEMPLATEDIR)/pg_ident.conf.sample
+	  $(DESTDIR)$(TEMPLATEDIR)/pg_ident.conf.sample
 	$(INSTALL) $(INSTLOPTS) optimizer/geqo/pg_geqo.sample \
-	  $(TEMPLATEDIR)/pg_geqo.sample
+	  $(DESTDIR)$(TEMPLATEDIR)/pg_geqo.sample
 	$(INSTALL) $(INSTLOPTS) pg_options.sample \
-	  $(TEMPLATEDIR)/pg_options.sample
+	  $(DESTDIR)$(TEMPLATEDIR)/pg_options.sample
 
 install-headers: fmgr.h $(SRCDIR)/include/config.h
-	-@if [ ! -d $(HEADERDIR) ]; then mkdir $(HEADERDIR); fi
-	-@if [ ! -d $(HEADERDIR)/port ]; then mkdir $(HEADERDIR)/port; fi
-	-@if [ ! -d $(HEADERDIR)/port/$(PORTNAME) ]; \
-		then mkdir $(HEADERDIR)/port/$(PORTNAME); fi
-	-@if [ ! -d $(HEADERDIR)/lib ]; \
-		then mkdir $(HEADERDIR)/lib; fi
-	-@if [ ! -d $(HEADERDIR)/libpq ]; \
-		then mkdir $(HEADERDIR)/libpq; fi
-	-@if [ ! -d $(HEADERDIR)/utils ]; \
-		then mkdir $(HEADERDIR)/utils; fi
-	-@if [ ! -d $(HEADERDIR)/access ]; \
-		then mkdir $(HEADERDIR)/access; fi
-	-@if [ ! -d $(HEADERDIR)/executor ]; \
-		then mkdir $(HEADERDIR)/executor; fi
-	-@if [ ! -d $(HEADERDIR)/commands ]; \
-		then mkdir $(HEADERDIR)/commands; fi
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)/port/$(PORTNAME)
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)/lib
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)/libpq
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)/utils
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)/access
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)/executor
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)/commands
 	$(INSTALL) $(INSTLOPTS) fmgr.h \
-          $(HEADERDIR)/fmgr.h
+          $(DESTDIR)$(HEADERDIR)/fmgr.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/os.h \
-          $(HEADERDIR)/os.h
+          $(DESTDIR)$(HEADERDIR)/os.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/config.h \
-          $(HEADERDIR)/config.h
+          $(DESTDIR)$(HEADERDIR)/config.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/c.h \
-          $(HEADERDIR)/c.h
+          $(DESTDIR)$(HEADERDIR)/c.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/postgres.h \
-          $(HEADERDIR)/postgres.h
+          $(DESTDIR)$(HEADERDIR)/postgres.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/postgres_ext.h \
-          $(HEADERDIR)/postgres_ext.h
+          $(DESTDIR)$(HEADERDIR)/postgres_ext.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/libpq/pqcomm.h \
-          $(HEADERDIR)/libpq/pqcomm.h
+          $(DESTDIR)$(HEADERDIR)/libpq/pqcomm.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/libpq/libpq-fs.h \
-          $(HEADERDIR)/libpq/libpq-fs.h
+          $(DESTDIR)$(HEADERDIR)/libpq/libpq-fs.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/lib/dllist.h \
-          $(HEADERDIR)/lib/dllist.h
+          $(DESTDIR)$(HEADERDIR)/lib/dllist.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/utils/geo_decls.h \
-          $(HEADERDIR)/utils/geo_decls.h
+          $(DESTDIR)$(HEADERDIR)/utils/geo_decls.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/utils/elog.h \
-          $(HEADERDIR)/utils/elog.h
+          $(DESTDIR)$(HEADERDIR)/utils/elog.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/utils/palloc.h \
-          $(HEADERDIR)/utils/palloc.h
+          $(DESTDIR)$(HEADERDIR)/utils/palloc.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/utils/mcxt.h \
-          $(HEADERDIR)/utils/mcxt.h
+          $(DESTDIR)$(HEADERDIR)/utils/mcxt.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/access/attnum.h \
-          $(HEADERDIR)/access/attnum.h
+          $(DESTDIR)$(HEADERDIR)/access/attnum.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/executor/spi.h \
-          $(HEADERDIR)/executor/spi.h
+          $(DESTDIR)$(HEADERDIR)/executor/spi.h
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/include/commands/trigger.h \
-          $(HEADERDIR)/commands/trigger.h
+          $(DESTDIR)$(HEADERDIR)/commands/trigger.h
 ifeq ($(PORTNAME), hpux)
 # is this still necessary?
 	$(INSTALL) $(INSTLOPTS) $(SRCDIR)/backend/port/hpux/fixade.h \
-          $(HEADERDIR)/port/hpux/fixade.h
-endif
-
-$(BINDIR):
-	mkdir $@
-
-$(LIBDIR):
-	mkdir $@
-
-$(HEADERDIR):
-	mkdir $@
-
-# TEMPLATEDIR is commonly the same as LIBDIR, but it might be different.
-# We don't want duplicate rules if they are the same.
-ifneq ($(TEMPLATEDIR),$(LIBDIR))
-
-$(TEMPLATEDIR):
-	mkdir $@
-
+          $(DESTDIR)$(HEADERDIR)/port/hpux/fixade.h
 endif
 
 
diff -Nru postgresql-7.0.2/src/bin/initdb/Makefile postgresql-7.0.2.new/src/bin/initdb/Makefile
--- postgresql-7.0.2/src/bin/initdb/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/initdb/Makefile	Thu Jul  6 10:01:02 2000
@@ -20,7 +20,8 @@
 	sed -e 's/__MULTIBYTE__/$(MULTIBYTE)/g' < initdb.sh > initdb
 
 install: initdb
-	$(INSTALL) $(INSTL_EXE_OPTS) $+ $(BINDIR)/$+
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) $+ $(DESTDIR)$(BINDIR)/$+
 
 clean:
 	rm -f initdb
diff -Nru postgresql-7.0.2/src/bin/initlocation/Makefile postgresql-7.0.2.new/src/bin/initlocation/Makefile
--- postgresql-7.0.2/src/bin/initlocation/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/initlocation/Makefile	Thu Jul  6 10:01:02 2000
@@ -20,7 +20,8 @@
 	cp -p initlocation.sh initlocation
 
 install: initlocation
-	$(INSTALL) $(INSTL_EXE_OPTS) $+ $(BINDIR)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) $+ $(DESTDIR)$(BINDIR)
 
 clean:
 	rm -f initlocation
diff -Nru postgresql-7.0.2/src/bin/ipcclean/Makefile postgresql-7.0.2.new/src/bin/ipcclean/Makefile
--- postgresql-7.0.2/src/bin/ipcclean/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/ipcclean/Makefile	Thu Jul  6 10:01:02 2000
@@ -19,7 +19,8 @@
 	cp $< $@
 
 install: ipcclean
-	$(INSTALL) $(INSTL_EXE_OPTS) $^ $(BINDIR)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) $^ $(DESTDIR)$(BINDIR)
 
 clean:
 	rm -f ipcclean
diff -Nru postgresql-7.0.2/src/bin/pg_ctl/Makefile postgresql-7.0.2.new/src/bin/pg_ctl/Makefile
--- postgresql-7.0.2/src/bin/pg_ctl/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/pg_ctl/Makefile	Thu Jul  6 10:01:02 2000
@@ -20,8 +20,10 @@
 	cat pg_ctl.sh > pg_ctl
 
 install: pg_ctl
-	$(INSTALL) $(INSTL_EXE_OPTS) $+ $(BINDIR)
-	$(INSTALL) $(INSTLOPTS) postmaster.opts.default.sample $(TEMPLATEDIR)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -d $(DESTDIR)$(TEMPLATEDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) $+ $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTLOPTS) postmaster.opts.default.sample $(DESTDIR)$(TEMPLATEDIR)
 
 clean:
 	rm -f pg_ctl
diff -Nru postgresql-7.0.2/src/bin/pg_dump/Makefile.in postgresql-7.0.2.new/src/bin/pg_dump/Makefile.in
--- postgresql-7.0.2/src/bin/pg_dump/Makefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/pg_dump/Makefile.in	Thu Jul  6 10:01:02 2000
@@ -39,9 +39,10 @@
 	$(MAKE) -C $(LIBPQDIR) libpq.a
 
 install: pg_dump
-	$(INSTALL) $(INSTL_EXE_OPTS) pg_dump$(X) $(BINDIR)/pg_dump$(X)
-	$(INSTALL) $(INSTL_EXE_OPTS) pg_dumpall $(BINDIR)/pg_dumpall
-	$(INSTALL) $(INSTL_EXE_OPTS) pg_upgrade $(BINDIR)/pg_upgrade
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) pg_dump$(X) $(DESTDIR)$(BINDIR)/pg_dump$(X)
+	$(INSTALL) $(INSTL_EXE_OPTS) pg_dumpall $(DESTDIR)$(BINDIR)/pg_dumpall
+	$(INSTALL) $(INSTL_EXE_OPTS) pg_upgrade $(DESTDIR)$(BINDIR)/pg_upgrade
 
 depend dep:
 	$(CC) -MM $(CFLAGS) *.c >depend
diff -Nru postgresql-7.0.2/src/bin/pg_encoding/Makefile postgresql-7.0.2.new/src/bin/pg_encoding/Makefile
--- postgresql-7.0.2/src/bin/pg_encoding/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/pg_encoding/Makefile	Thu Jul  6 10:01:02 2000
@@ -30,7 +30,8 @@
 	$(MAKE) -C $(LIBPQDIR) libpq.a
 
 install: pg_encoding
-	$(INSTALL) $(INSTL_EXE_OPTS) pg_encoding$(X) $(BINDIR)/pg_encoding$(X)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) pg_encoding$(X) $(DESTDIR)$(BINDIR)/pg_encoding$(X)
 
 depend dep:
 	$(CC) -MM $(CFLAGS) *.c >depend
diff -Nru postgresql-7.0.2/src/bin/pg_id/Makefile postgresql-7.0.2.new/src/bin/pg_id/Makefile
--- postgresql-7.0.2/src/bin/pg_id/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/pg_id/Makefile	Thu Jul  6 10:01:02 2000
@@ -20,7 +20,8 @@
 	$(CC) -o pg_id $(OBJS) $(LDFLAGS)
 
 install: pg_id
-	$(INSTALL) $(INSTL_EXE_OPTS) pg_id$(X) $(BINDIR)/pg_id$(X)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) pg_id$(X) $(DESTDIR)$(BINDIR)/pg_id$(X)
 
 depend dep:
 	$(CC) -MM $(CFLAGS) *.c >depend
diff -Nru postgresql-7.0.2/src/bin/pg_passwd/Makefile postgresql-7.0.2.new/src/bin/pg_passwd/Makefile
--- postgresql-7.0.2/src/bin/pg_passwd/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/pg_passwd/Makefile	Thu Jul  6 10:01:02 2000
@@ -15,7 +15,8 @@
 	$(CC) $(CFLAGS) -o pg_passwd $(OBJS) $(LDFLAGS)
 
 install: pg_passwd
-	$(INSTALL) $(INSTL_EXE_OPTS) pg_passwd$(X) $(BINDIR)/pg_passwd$(X)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) pg_passwd$(X) $(DESTDIR)$(BINDIR)/pg_passwd$(X)
 
 .PHONY: submake
 submake:
diff -Nru postgresql-7.0.2/src/bin/pg_version/Makefile.in postgresql-7.0.2.new/src/bin/pg_version/Makefile.in
--- postgresql-7.0.2/src/bin/pg_version/Makefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/pg_version/Makefile.in	Thu Jul  6 10:01:02 2000
@@ -27,7 +27,8 @@
 	$(MAKE) -C $(SRCDIR)/utils version.o
 
 install: pg_version
-	$(INSTALL) $(INSTL_EXE_OPTS) pg_version$(X) $(BINDIR)/pg_version$(X)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) pg_version$(X) $(DESTDIR)$(BINDIR)/pg_version$(X)
 
 depend dep:
 	$(CC) -MM $(CFLAGS) *.c >depend
diff -Nru postgresql-7.0.2/src/bin/pgaccess/Makefile postgresql-7.0.2.new/src/bin/pgaccess/Makefile
--- postgresql-7.0.2/src/bin/pgaccess/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/pgaccess/Makefile	Wed Jul 12 18:41:32 2000
@@ -21,17 +21,15 @@
 			> pgaccess
 
 install:
-	$(INSTALL) $(INSTL_EXE_OPTS) pgaccess $(BINDIR)/pgaccess
-	-@if [ ! -d $(POSTGRESDIR)/pgaccess ]; then mkdir $(POSTGRESDIR)/pgaccess; fi
-	$(INSTALL) $(INSTL_EXE_OPTS) main.tcl $(POSTGRESDIR)/pgaccess
-	-@if [ ! -d $(POSTGRESDIR)/pgaccess/lib ]; then mkdir $(POSTGRESDIR)/pgaccess/lib; fi
-	$(INSTALL) $(INSTLOPTS) lib/*.tcl $(POSTGRESDIR)/pgaccess/lib
-	-@if [ ! -d $(POSTGRESDIR)/pgaccess/lib/help ]; then mkdir $(POSTGRESDIR)/pgaccess/lib/help; fi
-	$(INSTALL) $(INSTLOPTS) lib/help/*.hlp $(POSTGRESDIR)/pgaccess/lib/help
-	-@if [ ! -d $(POSTGRESDIR)/pgaccess/lib/languages ]; then mkdir $(POSTGRESDIR)/pgaccess/lib/languages; fi
-	$(INSTALL) $(INSTLOPTS) lib/languages/[a-z]* $(POSTGRESDIR)/pgaccess/lib/languages
-	-@if [ ! -d $(POSTGRESDIR)/pgaccess/images ]; then mkdir $(POSTGRESDIR)/pgaccess/images; fi
-	$(INSTALL) $(INSTLOPTS) images/*.gif $(POSTGRESDIR)/pgaccess/images
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)/pgaccess/help
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)/pgaccess/anguages
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)/pgaccess/images
+	$(INSTALL) $(INSTL_EXE_OPTS) pgaccess $(DESTDIR)$(BINDIR)/pgaccess
+	$(INSTALL) $(INSTL_EXE_OPTS) main.tcl $(DESTDIR)$(LIBDIR)/pgaccess
+	$(INSTALL) $(INSTLOPTS) lib/*.tcl $(DESTDIR)$(LIBDIR)/pgaccess
+	$(INSTALL) $(INSTLOPTS) lib/help/*.hlp $(DESTDIR)$(LIBDIR)/pgaccess/help
+	$(INSTALL) $(INSTLOPTS) lib/languages/[a-z]* $(DESTDIR)$(LIBDIR)/pgaccess/languages
+	$(INSTALL) $(INSTLOPTS) images/*.gif $(DESTDIR)$(LIBDIR)/pgaccess/images
 
 clean:
 	rm -f pgaccess
diff -Nru postgresql-7.0.2/src/bin/pgtclsh/Makefile postgresql-7.0.2.new/src/bin/pgtclsh/Makefile
--- postgresql-7.0.2/src/bin/pgtclsh/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/pgtclsh/Makefile	Thu Jul  6 10:01:02 2000
@@ -53,10 +53,12 @@
 install: $(INSTPGMS)
 
 install_tcl: pgtclsh
-	$(INSTALL) $(INSTL_EXE_OPTS) pgtclsh $(BINDIR)/pgtclsh
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) pgtclsh $(DESTDIR)$(BINDIR)/pgtclsh
 
 install_tk: pgtksh
-	$(INSTALL) $(INSTL_EXE_OPTS) pgtksh $(BINDIR)/pgtksh
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) pgtksh $(DESTDIR)$(BINDIR)/pgtksh
 
 Makefile.tcldefs: mkMakefile.tcldefs.sh
 	/bin/sh mkMakefile.tcldefs.sh
diff -Nru postgresql-7.0.2/src/bin/psql/Makefile.in postgresql-7.0.2.new/src/bin/psql/Makefile.in
--- postgresql-7.0.2/src/bin/psql/Makefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/psql/Makefile.in	Thu Jul  6 10:01:02 2000
@@ -76,7 +76,8 @@
 	$(MAKE) -C $(LIBPQDIR) libpq.a
 
 install: psql
-	$(INSTALL) $(INSTL_EXE_OPTS) psql$(X) $(BINDIR)/psql$(X)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) psql$(X) $(DESTDIR)$(BINDIR)/psql$(X)
 
 depend dep:
 	$(CC) -MM -MG $(CFLAGS) *.c >depend
diff -Nru postgresql-7.0.2/src/bin/scripts/Makefile postgresql-7.0.2.new/src/bin/scripts/Makefile
--- postgresql-7.0.2/src/bin/scripts/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/bin/scripts/Makefile	Thu Jul  6 10:01:02 2000
@@ -32,7 +32,8 @@
 vacuumdb:
 
 install: $(SCRIPTS)
-	for i in $+; do $(INSTALL) $(INSTL_EXE_OPTS) $$i $(BINDIR); done
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	for i in $+; do $(INSTALL) $(INSTL_EXE_OPTS) $$i $(DESTDIR)$(BINDIR); done
 
 clean:
 	rm -f createlang
diff -Nru postgresql-7.0.2/src/interfaces/ecpg/include/Makefile postgresql-7.0.2.new/src/interfaces/ecpg/include/Makefile
--- postgresql-7.0.2/src/interfaces/ecpg/include/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/interfaces/ecpg/include/Makefile	Thu Jul  6 10:01:02 2000
@@ -6,11 +6,12 @@
 	@echo Nothing to be done.
 
 install::
-	$(INSTALL) $(INSTLOPTS) ecpgerrno.h $(HEADERDIR)	
-	$(INSTALL) $(INSTLOPTS) ecpglib.h $(HEADERDIR)	
-	$(INSTALL) $(INSTLOPTS) ecpgtype.h $(HEADERDIR)	
-	$(INSTALL) $(INSTLOPTS) sqlca.h $(HEADERDIR)	
-	$(INSTALL) $(INSTLOPTS) sql3types.h $(HEADERDIR)	
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)
+	$(INSTALL) $(INSTLOPTS) ecpgerrno.h $(DESTDIR)$(HEADERDIR)	
+	$(INSTALL) $(INSTLOPTS) ecpglib.h $(DESTDIR)$(HEADERDIR)	
+	$(INSTALL) $(INSTLOPTS) ecpgtype.h $(DESTDIR)$(HEADERDIR)	
+	$(INSTALL) $(INSTLOPTS) sqlca.h $(DESTDIR)$(HEADERDIR)	
+	$(INSTALL) $(INSTLOPTS) sql3types.h $(DESTDIR)$(HEADERDIR)	
 
 uninstall::
 	rm -f $(HEADERDIR)/ecpgerrno.h
diff -Nru postgresql-7.0.2/src/interfaces/ecpg/preproc/Makefile.in postgresql-7.0.2.new/src/interfaces/ecpg/preproc/Makefile.in
--- postgresql-7.0.2/src/interfaces/ecpg/preproc/Makefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/interfaces/ecpg/preproc/Makefile.in	Thu Jul  6 10:01:02 2000
@@ -42,7 +42,8 @@
 # to ship those files in the distribution, for people with inadequate tools.
 
 install: all
-	$(INSTALL) $(INSTL_EXE_OPTS) ecpg$(X) $(BINDIR)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) $(INSTL_EXE_OPTS) ecpg$(X) $(DESTDIR)$(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/ecpg
diff -Nru postgresql-7.0.2/src/interfaces/libpgeasy/Makefile.in postgresql-7.0.2.new/src/interfaces/libpgeasy/Makefile.in
--- postgresql-7.0.2/src/interfaces/libpgeasy/Makefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/interfaces/libpgeasy/Makefile.in	Thu Jul  6 10:01:02 2000
@@ -39,8 +39,8 @@
 install: install-headers install-lib $(install-shlib-dep)
 
 install-headers: libpgeasy.h
-	-@if [ ! -d $(HEADERDIR) ]; then mkdir $(HEADERDIR); fi
-	$(INSTALL) $(INSTLOPTS) libpgeasy.h $(HEADERDIR)/libpgeasy.h
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)
+	$(INSTALL) $(INSTLOPTS) libpgeasy.h $(DESTDIR)$(HEADERDIR)/libpgeasy.h
 
 .PHONY: clean
 
diff -Nru postgresql-7.0.2/src/interfaces/libpgtcl/Makefile.in postgresql-7.0.2.new/src/interfaces/libpgtcl/Makefile.in
--- postgresql-7.0.2/src/interfaces/libpgtcl/Makefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/interfaces/libpgtcl/Makefile.in	Thu Jul  6 10:01:02 2000
@@ -43,11 +43,8 @@
 install: install-headers install-lib $(install-shlib-dep)
 
 install-headers: beforeinstall-headers libpgtcl.h
-	$(INSTALL) $(INSTLOPTS) libpgtcl.h $(HEADERDIR)/libpgtcl.h
-
-beforeinstall-headers:
-	-@if [ ! -d $(HEADERDIR) ]; then mkdir $(HEADERDIR); fi
-
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)
+	$(INSTALL) $(INSTLOPTS) libpgtcl.h $(DESTDIR)$(HEADERDIR)/libpgtcl.h
 
 .PHONY: clean
 
diff -Nru postgresql-7.0.2/src/interfaces/libpq/Makefile.in postgresql-7.0.2.new/src/interfaces/libpq/Makefile.in
--- postgresql-7.0.2/src/interfaces/libpq/Makefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/interfaces/libpq/Makefile.in	Thu Jul  6 10:01:02 2000
@@ -83,11 +83,10 @@
 install: install-headers install-lib $(install-shlib-dep)
 
 install-headers: libpq-fe.h libpq-int.h
-	-@if [ ! -d $(HEADERDIR) ]; then mkdir $(HEADERDIR); fi
-	$(INSTALL) $(INSTLOPTS) libpq-fe.h $(HEADERDIR)/libpq-fe.h
-	$(INSTALL) $(INSTLOPTS) libpq-int.h $(HEADERDIR)/libpq-int.h
-	$(INSTALL) $(INSTLOPTS) pqexpbuffer.h $(HEADERDIR)/pqexpbuffer.h
-
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)
+	$(INSTALL) $(INSTLOPTS) libpq-fe.h $(DESTDIR)$(HEADERDIR)/libpq-fe.h
+	$(INSTALL) $(INSTLOPTS) libpq-int.h $(DESTDIR)$(HEADERDIR)/libpq-int.h
+	$(INSTALL) $(INSTLOPTS) pqexpbuffer.h $(DESTDIR)$(HEADERDIR)/pqexpbuffer.h
 
 .PHONY: clean
 
diff -Nru postgresql-7.0.2/src/interfaces/libpq++/Makefile.in postgresql-7.0.2.new/src/interfaces/libpq++/Makefile.in
--- postgresql-7.0.2/src/interfaces/libpq++/Makefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/interfaces/libpq++/Makefile.in	Thu Jul  6 10:01:02 2000
@@ -73,16 +73,13 @@
 		 pgcursordb.h \
 		 pglobject.h
 
-install-headers: beforeinstall-headers $(MAINHEADER)
-	@$(INSTALL) $(INSTLOPTS) $(MAINHEADER) $(HEADERDIR)/$(MAINHEADER)
+install-headers: $(MAINHEADER)
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR) $(DESTDIR)$(LIBPGXXHEADERDIR)
+	@$(INSTALL) $(MAINHEADER) $(DESTDIR)$(HEADERDIR)/$(MAINHEADER)
 	@for i in ${LIBPGXXHEADERS}; do \
 		echo "Installing $(LIBPGXXHEADERDIR)/$$i."; \
-		$(INSTALL) $(INSTLOPTS) $$i $(LIBPGXXHEADERDIR)/$$i; \
+		$(INSTALL) $(INSTLOPTS) $$i $(DESTDIR)$(LIBPGXXHEADERDIR)/$$i; \
 	done
-
-beforeinstall-headers:
-	-@if [ ! -d $(HEADERDIR) ]; then mkdir $(HEADERDIR); fi
-	-@if [ ! -d $(LIBPGXXHEADERDIR) ]; then mkdir $(LIBPGXXHEADERDIR); fi
 
 .PHONY: clean
 
diff -Nru postgresql-7.0.2/src/interfaces/odbc/GNUmakefile.in postgresql-7.0.2.new/src/interfaces/odbc/GNUmakefile.in
--- postgresql-7.0.2/src/interfaces/odbc/GNUmakefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/interfaces/odbc/GNUmakefile.in	Mon Jul 24 10:57:29 2000
@@ -15,6 +15,7 @@
 NAME = psqlodbc
 SRCDIR= @top_srcdir@
 ODBCSRCDIR= @srcdir@
+sysconfdir= @sysconfdir@
 include $(SRCDIR)/Makefile.global
 
 include Version.mk
@@ -46,22 +47,17 @@
 
 .PHONY: install install-ini beforeinstall-headers install-headers
 
-install: $(HEADERDIR) $(LIBDIR) $(ODBCINST) install-headers \
-	install-ini install-lib $(install-shlib-dep)
+install: install-headers install-ini install-lib $(install-shlib-dep)
 
-$(HEADERDIR) $(LIBDIR) $(ODBCINST):
-	mkdir -p $@
-
-install-headers: beforeinstall-headers isql.h isqlext.h iodbc.h
-	$(INSTALL) $(INSTLOPTS) iodbc.h $(HEADERDIR)/iodbc/iodbc.h
-	$(INSTALL) $(INSTLOPTS) isql.h $(HEADERDIR)/iodbc/isql.h
-	$(INSTALL) $(INSTLOPTS) isqlext.h $(HEADERDIR)/iodbc/isqlext.h
-
-beforeinstall-headers:
-	@if [ ! -d $(HEADERDIR)/iodbc ]; then mkdir -p $(HEADERDIR)/iodbc; fi
+install-headers: isql.h isqlext.h iodbc.h
+	$(INSTALL) -d $(DESTDIR)$(HEADERDIR)/iodbc
+	$(INSTALL) $(INSTLOPTS) iodbc.h $(DESTDIR)$(HEADERDIR)/iodbc/iodbc.h
+	$(INSTALL) $(INSTLOPTS) isql.h $(DESTDIR)$(HEADERDIR)/iodbc/isql.h
+	$(INSTALL) $(INSTLOPTS) isqlext.h $(DESTDIR)$(HEADERDIR)/iodbc/isqlext.h
 
 install-ini: odbcinst.ini
-	$(INSTALL) $(INSTL_LIB_OPTS) odbcinst.ini $(ODBCINST)/odbcinst.ini
+	$(INSTALL) -d $(DESTDIR)$(sysconfdir)
+	$(INSTALL) $(INSTL_LIB_OPTS) odbcinst.ini $(DESTDIR)$(sysconfdir)/odbcinst.ini
 
 .PHONY: clean
 
diff -Nru postgresql-7.0.2/src/interfaces/odbc/Makefile.global.in postgresql-7.0.2.new/src/interfaces/odbc/Makefile.global.in
--- postgresql-7.0.2/src/interfaces/odbc/Makefile.global.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/interfaces/odbc/Makefile.global.in	Sat Jul 22 08:10:28 2000
@@ -78,11 +78,11 @@
 #POSTDOCDIR= $(POSTGRESDIR)/doc
 
 # Where the header files necessary to build frontend programs get installed.
-HEADERDIR= $(POSTGRESDIR)/include
+HEADERDIR= @includedir@/pgsql
 
 # Where the odbcinst.ini file will be placed
 
-ODBCINST= @ODBCINST@
+sysconfdir= @sysconfdir@
 
 ##############################################################################
 #
diff -Nru postgresql-7.0.2/src/pl/plpgsql/src/Makefile.in postgresql-7.0.2.new/src/pl/plpgsql/src/Makefile.in
--- postgresql-7.0.2/src/pl/plpgsql/src/Makefile.in	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/pl/plpgsql/src/Makefile.in	Wed Jul 12 18:42:56 2000
@@ -40,7 +40,8 @@
 
 install: $(shlib)
 ifneq ($(shlib),)
-	$(INSTALL) $(INSTL_SHLIB_OPTS) $(shlib) $(LIBDIR)/plpgsql$(DLSUFFIX)
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)
+	$(INSTALL) $(INSTL_SHLIB_OPTS) $(shlib) $(DESTDIR)$(LIBDIR)/plpgsql$(DLSUFFIX)
 else
 	@echo "plpgsql not installed due to lack of shared library support."
 endif
diff -Nru postgresql-7.0.2/src/pl/tcl/Makefile postgresql-7.0.2.new/src/pl/tcl/Makefile
--- postgresql-7.0.2/src/pl/tcl/Makefile	Mon Jul 24 10:56:31 2000
+++ postgresql-7.0.2.new/src/pl/tcl/Makefile	Wed Jul 12 18:44:19 2000
@@ -103,7 +103,8 @@
 all: $(INFILES)
 
 install: all
-	$(INSTALL) $(INSTL_SHLIB_OPTS) $(DLOBJS) $(LIBDIR)/$(DLOBJS)
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)
+	$(INSTALL) $(INSTL_SHLIB_OPTS) $(DLOBJS) $(DESTDIR)$(LIBDIR)/$(DLOBJS)
 
 else
 
