diff -urN postgresql-7.1.org/src/pl/plperl/GNUmakefile postgresql-7.1/src/pl/plperl/GNUmakefile
--- postgresql-7.1.org/src/pl/plperl/GNUmakefile	Wed Apr 18 13:45:31 2001
+++ postgresql-7.1/src/pl/plperl/GNUmakefile	Wed Apr 18 13:46:08 2001
@@ -16,7 +16,7 @@
 	$(PERL) $<
 
 install: all installdirs
-	$(MAKE) -f Makefile install
+	$(MAKE) -f Makefile install DESTDIR=\"$(DESTDIR)\"
 
 installdirs:
 	$(mkinstalldirs) $(DESTDIR)$(plperl_installdir)
