--- postgresql-8.0.0beta2/src/backend/utils/misc/postgresql.conf.sample	2004-08-12 21:03:36.000000000 +0200
+++ postgresql-8.0.0beta2/src/backend/utils/misc/postgresql.conf.sample.new	2004-09-05 20:54:31.087203264 +0200
@@ -164,7 +164,7 @@
 
 # - Where to Log -
 
-#log_destination = 'stderr'	# Valid values are combinations of stderr,
+log_destination = 'syslog'	# Valid values are combinations of stderr,
                                 # syslog and eventlog, depending on
                                 # platform.
 
@@ -180,8 +180,8 @@
                             # so many kilobytes of log output.  0 to disable.
 
 # These are relevant when logging to syslog:
-#syslog_facility = 'LOCAL0'
-#syslog_ident = 'postgres'
+syslog_facility = 'LOCAL0'
+syslog_ident = 'postgres'
 
 
 # - When to Log -
@@ -211,10 +211,10 @@
 #debug_print_rewritten = false
 #debug_print_plan = false
 #debug_pretty_print = false
-#log_connections = false
+log_connections = true
 #log_disconnections = false
 #log_duration = false
-#log_line_prefix = ''		# e.g. '<%u%%%d> ' 
+log_line_prefix = '<%p> '	# e.g. '<%u%%%d> ' 
 				# %u=user name %d=database name
 				# %r=remote host and port
 				# %p=PID %t=timestamp %i=command tag
