diff -Nur postgresql-7.3.2.orig/configure.in postgresql-7.3.2/configure.in
--- postgresql-7.3.2.orig/configure.in	Sun Feb  2 20:56:45 2003
+++ postgresql-7.3.2/configure.in	Wed Feb  5 11:25:49 2003
@@ -20,9 +20,7 @@
 dnl The GNU folks apparently haven't heard that some people don't use
 dnl Texinfo. Use this sorcery to use "docdir" instead of "infodir".
 m4_define([info], [doc])
-m4_define([infodir], [docdir])
 AC_INIT([PostgreSQL], [7.4beta4], [pgsql-bugs@postgresql.org])
-m4_undefine([infodir])
 m4_undefine([info])
 AC_SUBST(docdir)
 
@@ -34,6 +32,8 @@
 AC_SUBST(configure_args, [$ac_configure_args])
 
 AC_DEFINE_UNQUOTED(PG_VERSION, "$PACKAGE_VERSION", [PostgreSQL version])
+docdir=$infodir
+AC_SUBST(docdir)
 
 AC_CANONICAL_HOST
 
