diff -Naur postgresql-7.4.2-orig/configure.in postgresql-7.4.2/configure.in
--- postgresql-7.4.2-orig/configure.in	2004-03-05 19:57:20.000000000 +0000
+++ postgresql-7.4.2/configure.in	2004-03-11 06:53:31.000000000 +0000
@@ -20,9 +20,7 @@
 dnl The GNU folks apparently haven't heard that some people don't use
 dnl Texinfo. Use this sorcery to use "docdir" instead of "infodir".
 m4_define([info], [doc])
-m4_define([infodir], [docdir])
 AC_INIT([PostgreSQL], [7.4.2], [pgsql-bugs@postgresql.org])
-m4_undefine([infodir])
 m4_undefine([info])
 AC_SUBST(docdir)
 
