diff -urNp -x '*.orig' postgresql-11.5.org/src/backend/utils/misc/postgresql.conf.sample postgresql-11.5/src/backend/utils/misc/postgresql.conf.sample
--- postgresql-11.5.org/src/backend/utils/misc/postgresql.conf.sample	2019-08-05 23:14:59.000000000 +0200
+++ postgresql-11.5/src/backend/utils/misc/postgresql.conf.sample	2021-05-08 22:44:29.704690750 +0200
@@ -361,7 +361,7 @@
 
 # - Where to Log -
 
-#log_destination = 'stderr'		# Valid values are combinations of
+log_destination = 'syslog'		# Valid values are combinations of
 					# stderr, csvlog, syslog, and eventlog,
 					# depending on platform.  csvlog
 					# requires logging_collector to be on.
@@ -394,9 +394,9 @@
 					# 0 disables.
 
 # These are relevant when logging to syslog:
-#syslog_facility = 'LOCAL0'
-#syslog_ident = 'postgres'
-#syslog_sequence_numbers = on
+syslog_facility = 'LOCAL0'
+syslog_ident = 'postgres'
+syslog_sequence_numbers = on
 #syslog_split_messages = on
 
 # This is only relevant when logging to eventlog (win32):
