--- postgresql-7.4/contrib/contrib-global.mk.orig	2003-12-17 16:54:15.000000000 +0100
+++ postgresql-7.4/contrib/contrib-global.mk	2003-12-17 16:55:54.000000000 +0100
@@ -74,8 +74,8 @@
 install: all installdirs
 ifneq (,$(DATA)$(DATA_built))
 	@for file in $(addprefix $(srcdir)/, $(DATA)) $(DATA_built); do \
-	  echo "$(INSTALL_DATA) $$file $(DESTDIR)$(datadir)/contrib"; \
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(datadir)/contrib; \
+	  echo "$(INSTALL_DATA) $$file $(DESTDIR)$(datadir)"; \
+	  $(INSTALL_DATA) $$file $(DESTDIR)$(datadir); \
 	done
 endif # DATA
 ifdef MODULES
@@ -84,12 +84,6 @@
 	  $(INSTALL_SHLIB) $$file $(DESTDIR)$(pkglibdir); \
 	done
 endif # MODULES
-ifdef DOCS
-	@for file in $(addprefix $(srcdir)/, $(DOCS)); do \
-	  echo "$(INSTALL_DATA) $$file $(DESTDIR)$(docdir)/contrib"; \
-	  $(INSTALL_DATA) $$file $(DESTDIR)$(docdir)/contrib; \
-	done
-endif # DOCS
 ifdef PROGRAM
 	$(INSTALL_PROGRAM) $(PROGRAM)$(X) $(DESTDIR)$(bindir)
 endif # PROGRAM
