--- postgresql-7.4RC2/src/interfaces/ecpg/include/Makefile.orig	2003-10-20 14:54:29.000000000 +0200
+++ postgresql-7.4RC2/src/interfaces/ecpg/include/Makefile	2003-11-16 21:26:36.294013624 +0100
@@ -8,10 +8,10 @@
 ecpg_headers = ecpgerrno.h ecpglib.h ecpgtype.h sqlca.h sql3types.h ecpg_informix.h pgtypes_error.h pgtypes_numeric.h pgtypes_timestamp.h pgtypes_date.h datetime.h decimal.h pgtypes_interval.h sqltypes.h compatlib.h sqlda.h
 
 install-headers: $(ecpg_headers)
-	for i in $^; do $(INSTALL_DATA) $$i $(DESTDIR)$(includedir); done
+	for i in $^; do $(INSTALL_DATA) $$i $(DESTDIR)$(includedir)/ecpg; done
 
 installdirs:
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
+	$(mkinstalldirs) $(DESTDIR)$(includedir)/ecpg
 
 uninstall:
 	rm -f $(addprefix $(DESTDIR)$(includedir)/, $(ecpg_headers))
