diff -ur postgresql-7.1.3/src/bin/pg_ctl/pg_ctl.sh postgresql-7.1.3.new/src/bin/pg_ctl/pg_ctl.sh
--- postgresql-7.1.3/src/bin/pg_ctl/pg_ctl.sh	Sat Apr 21 13:23:58 2001
+++ postgresql-7.1.3.new/src/bin/pg_ctl/pg_ctl.sh	Sat Dec  1 11:02:22 2001
@@ -295,7 +295,7 @@
 if [ $op = "start" -o $op = "restart" ];then
     oldpid=""
     if [ -f $PIDFILE ];then
-	echo "$CMDNAME: Another postmaster may be running.  Trying to start postmaster anyway." 1>&2
+	$silence_echo echo "$CMDNAME: Another postmaster may be running.  Trying to start postmaster anyway." 1>&2
 	oldpid=`sed -n 1p $PIDFILE`
     fi
 
