diff -ur postgresql-6.4.2.old/src/Makefile.global.in postgresql-6.4.2/src/Makefile.global.in
--- postgresql-6.4.2.old/src/Makefile.global.in	Wed Jun 30 10:43:20 1999
+++ postgresql-6.4.2/src/Makefile.global.in	Wed Jun 30 10:42:13 1999
@@ -62,13 +62,13 @@
 
 # Where the postgres executables live (changeable by just putting them
 # somewhere else and putting that directory in your shell PATH)
-BINDIR= $(POSTGRESDIR)/bin
+BINDIR= $(DESTDIR)$(POSTGRESDIR)/bin
 
 # Where libpq.a gets installed.  You must put it where your loader will
 # look for it if you wish to use the -lpq convention.  Otherwise you
 # can just put the absolute pathname to the library at the end of your
 # command line.
-LIBDIR= $(POSTGRESDIR)/lib
+LIBDIR= $(DESTDIR)$(POSTGRESDIR)/lib
 
 # This is the directory where IPC utilities ipcs and ipcrm are located
 #
@@ -76,18 +76,18 @@
 IPCRM=@ipcrm@ 
 
 # Where the man pages (suitable for use with "man") get installed.
-POSTMANDIR= $(POSTGRESDIR)/man
+POSTMANDIR= $(DESTDIR)@mandir@
 
 # Where the formatted documents (e.g., the reference manual) get installed.
-POSTDOCDIR= $(POSTGRESDIR)/doc
+POSTDOCDIR= $(DESTDIR)$(POSTGRESDIR)/doc
 
 # Where the header files necessary to build frontend programs get installed.
-HEADERDIR= $(POSTGRESDIR)/include
+HEADERDIR= $(DESTDIR)$(POSTGRESDIR)/include
 
 # Where the ODBC installation-wide configuration file gets installed.
 # This needs to be a client-side readable area, so can't go in $PGDATA.
 # - thomas 1998-10-05
-ODBCINST= $(POSTGRESDIR)
+ODBCINST= $(DESTDIR)$(POSTGRESDIR)
 
 ##############################################################################
 #
