diff -urNp -x '*.orig' postgresql-12.4.org/src/backend/utils/misc/postgresql.conf.sample postgresql-12.4/src/backend/utils/misc/postgresql.conf.sample
--- postgresql-12.4.org/src/backend/utils/misc/postgresql.conf.sample	2020-08-10 23:15:53.000000000 +0200
+++ postgresql-12.4/src/backend/utils/misc/postgresql.conf.sample	2021-11-09 09:30:59.882648111 +0100
@@ -416,7 +416,7 @@
 
 # - Where to Log -
 
-#log_destination = 'stderr'		# Valid values are combinations of
+log_destination = 'syslog'		# Valid values are combinations of
 					# stderr, csvlog, syslog, and eventlog,
 					# depending on platform.  csvlog
 					# requires logging_collector to be on.
@@ -449,9 +449,9 @@
 					# 0 disables.
 
 # These are relevant when logging to syslog:
-#syslog_facility = 'LOCAL0'
-#syslog_ident = 'postgres'
-#syslog_sequence_numbers = on
+syslog_facility = 'LOCAL0'
+syslog_ident = 'postgres'
+syslog_sequence_numbers = on
 #syslog_split_messages = on
 
 # This is only relevant when logging to eventlog (win32):
