--- src/bin/pg_ctl/pg_ctl.sh.orig	2003-08-05 01:59:39.000000000 +0200
+++ src/bin/pg_ctl/pg_ctl.sh	2003-08-08 22:04:34.000000000 +0200
@@ -98,12 +98,12 @@
 fi
 
 # Check if needed programs actually exist in path
-if [ -x "$self_path/postmaster" ] && [ -x "$self_path/psql" ]; then
+if [ -x "$self_path/postmaster" ]; then
     PGPATH="$self_path"
-elif [ -x "$bindir/postmaster" ] && [ -x "$bindir/psql" ]; then
+elif [ -x "$bindir/postmaster" ]; then
     PGPATH="$bindir"
 else
-    echo "The programs \"postmaster\" and \"psql\" are needed by $CMDNAME but" 1>&2
+    echo "The programs \"postmaster\" is needed by $CMDNAME but" 1>&2
     echo "were not found in the directory \"$bindir\"." 1>&2
     echo "Check your installation." 1>&2
     exit 1
