diff -Nur postgresql-7.4RC1.orig/configure.in postgresql-7.4RC1/configure.in
--- postgresql-7.4RC1.orig/configure.in	2003-11-03 06:27:18.000000000 +0100
+++ postgresql-7.4RC1/configure.in	2003-11-07 12:03:08.000000000 +0100
@@ -783,7 +783,7 @@
 
 if test "$with_krb5" = yes ; then
   AC_CHECK_HEADER(krb5.h, [], [AC_MSG_ERROR([header file <krb5.h> is required for Kerberos 5])])
-  AC_CHECK_HEADER(com_err.h, [], [AC_MSG_ERROR([header file <com_err.h> is required for Kerberos 5])])
+  AC_CHECK_HEADER(et/com_err.h, [], [AC_MSG_ERROR([header file <et/com_err.h> is required for Kerberos 5])])
 fi
 
 if test "$with_openssl" = yes ; then
diff -Nur postgresql-7.4RC1.orig/src/backend/libpq/auth.c postgresql-7.4RC1/src/backend/libpq/auth.c
--- postgresql-7.4RC1.orig/src/backend/libpq/auth.c	2003-09-26 17:27:31.000000000 +0200
+++ postgresql-7.4RC1/src/backend/libpq/auth.c	2003-11-07 12:04:47.000000000 +0100
@@ -150,7 +150,7 @@
  */
 
 #include <krb5.h>
-#include <com_err.h>
+#include <et/com_err.h>
 
 /*
  * pg_an_to_ln -- return the local name corresponding to an authentication
diff -Nur postgresql-7.4RC1.orig/src/interfaces/libpq/fe-auth.c postgresql-7.4RC1/src/interfaces/libpq/fe-auth.c
--- postgresql-7.4RC1.orig/src/interfaces/libpq/fe-auth.c	2003-10-25 05:48:47.000000000 +0200
+++ postgresql-7.4RC1/src/interfaces/libpq/fe-auth.c	2003-11-07 12:04:52.000000000 +0100
@@ -243,7 +243,7 @@
  */
 
 #include <krb5.h>
-#include <com_err.h>
+#include <et/com_err.h>
 
 /*
  * pg_an_to_ln -- return the local name corresponding to an authentication
