--- postgresql-snapshot/src/backend/utils/misc/postgresql.conf.sample.orig	2007-09-05 19:10:48.000000000 +0100
+++ postgresql-snapshot/src/backend/utils/misc/postgresql.conf.sample	2007-09-08 18:25:28.757329916 +0100
@@ -228,7 +228,7 @@
 
 # - Where to Log -
 
-#log_destination = 'stderr'		# Valid values are combinations of 
+log_destination = 'syslog'		# Valid values are combinations of 
 					# stderr, csvlog, syslog and eventlog, 
 					# depending on platform.
 					# csvlog requires logging_collector to be on
@@ -259,8 +259,8 @@
 					# output.  0 to disable.
 
 # These are relevant when logging to syslog:
-#syslog_facility = 'LOCAL0'
-#syslog_ident = 'postgres'
+syslog_facility = 'LOCAL0'
+syslog_ident = 'postgres'
 
 
 # - When to Log -
@@ -322,7 +322,7 @@
 #debug_print_plan = off
 #debug_pretty_print = off
 #log_checkpoints = off
-#log_connections = off
+log_connections = on
 #log_disconnections = off
 #log_duration = off
 #log_hostname = off
