--- postgresql-7.4.6/configure.in.orig	2004-10-22 02:26:22.000000000 +0200
+++ postgresql-7.4.6/configure.in	2004-10-23 10:53:26.446102128 +0200
@@ -20,9 +20,7 @@
 dnl The GNU folks apparently haven't heard that some people don't use
 dnl Texinfo. Use this sorcery to use "docdir" instead of "infodir".
 m4_define([info], [doc])
-m4_define([infodir], [docdir])
 AC_INIT([PostgreSQL], [7.4.6], [pgsql-bugs@postgresql.org])
-m4_undefine([infodir])
 m4_undefine([info])
 AC_SUBST(docdir)
 
@@ -34,6 +32,8 @@
 AC_SUBST(configure_args, [$ac_configure_args])
 
 AC_DEFINE_UNQUOTED(PG_VERSION, "$PACKAGE_VERSION", [PostgreSQL version])
+docdir=$infodir
+AC_SUBST(docdir)
 
 AC_CANONICAL_HOST
 
--- postgresql-7.4/doc/Makefile.orig	2003-12-18 09:05:55.000000000 +0100
+++ postgresql-7.4/doc/Makefile	2003-12-18 09:07:05.000000000 +0100
@@ -25,9 +25,6 @@
 
 .NOTPARALLEL:
 
-ifneq ($(wildcard $(srcdir)/postgres.tar.gz),)
-found_html := yes
-endif
 
 ifneq ($(wildcard $(srcdir)/man.tar.gz),)
 # SCO OpenServer's man system is sufficiently different to not bother.
