--- postgresql-7.4.1/src/interfaces/ecpg/include/Makefile.orig	2003-12-18 19:55:05.000000000 +0100
+++ postgresql-7.4.1/src/interfaces/ecpg/include/Makefile	2003-12-23 18:57:34.712017440 +0100
@@ -12,11 +12,11 @@
 informix_headers = datetime.h decimal.h sqltypes.h sqlda.h
 
 install-headers: $(ecpg_headers) $(informix_headers)
-	for i in $(ecpg_headers); do $(INSTALL_DATA) $(srcdir)/$$i $(DESTDIR)$(includedir) || exit; done
-	for i in $(informix_headers); do $(INSTALL_DATA) $(srcdir)/$$i $(DESTDIR)$(informix_esql_dir) || exit; done
+	for i in $(ecpg_headers); do $(INSTALL_DATA) $(srcdir)/$$i $(DESTDIR)$(includedir)/ecpg || exit; done
+	for i in $(informix_headers); do $(INSTALL_DATA) $(srcdir)/$$i $(DESTDIR)$(includedir)/ecpg || exit; done
 
 installdirs:
-	$(mkinstalldirs) $(DESTDIR)$(includedir) $(DESTDIR)$(informix_esql_dir)
+	$(mkinstalldirs) $(DESTDIR)$(includedir)/ecpg
 
 uninstall:
 	rm -f $(addprefix $(DESTDIR)$(includedir)/, $(ecpg_headers))
