--- postgresql-7.4.1/src/interfaces/ecpg/include/Makefile.orig	2003-12-18 19:55:05.000000000 +0100
+++ postgresql-7.4.1/src/interfaces/ecpg/include/Makefile	2003-12-23 18:57:34.712017440 +0100
@@ -19,12 +19,12 @@
 informix_headers = datetime.h decimal.h sqltypes.h
 
 install-headers: $(ecpg_headers) $(informix_headers)
-	$(INSTALL_DATA) $(addprefix $(srcdir)/,$(ecpg_headers)) '$(DESTDIR)$(includedir)/'
-	$(INSTALL_DATA) $(addprefix $(srcdir)/,$(informix_headers)) '$(DESTDIR)$(informix_esql_dir)/'
+	$(INSTALL_DATA) $(addprefix $(srcdir)/,$(ecpg_headers)) '$(DESTDIR)$(includedir)/ecpg'
+	$(INSTALL_DATA) $(addprefix $(srcdir)/,$(informix_headers)) '$(DESTDIR)$(includedir)/ecpg'
 	$(INSTALL_DATA) $(ecpg_config_h) '$(DESTDIR)$(includedir)'
 
 installdirs:
-	$(MKDIR_P) '$(DESTDIR)$(includedir)' '$(DESTDIR)$(informix_esql_dir)'
+	$(MKDIR_P) '$(DESTDIR)$(includedir)/ecpg'
 
 uninstall:
 	rm -f $(addprefix '$(DESTDIR)$(includedir)'/, $(ecpg_headers))

