diff -urN postgresql-7.1.org/configure.in postgresql-7.1/configure.in
--- postgresql-7.1.org/configure.in	Wed Apr 18 13:50:14 2001
+++ postgresql-7.1/configure.in	Wed Apr 18 13:50:20 2001
@@ -661,7 +661,6 @@
 AC_CHECK_LIB(m,        main)
 AC_CHECK_LIB(dl,       main)
 AC_CHECK_LIB(socket,   main)
-AC_CHECK_LIB(nsl,      main)
 AC_CHECK_LIB(ipc,      main)
 AC_CHECK_LIB(IPC,      main)
 AC_CHECK_LIB(lc,       main)
