--- postgresql-7.4beta1/configure.in.orig	2003-08-05 00:30:30.000000000 +0200
+++ postgresql-7.4beta1/configure.in	2003-08-09 08:01:18.000000000 +0200
@@ -614,12 +614,13 @@
 
 if test "$PORTNAME" != "aix" -a "$PORTNAME" != "alpha"
 then
-	AC_CHECK_LIB(bsd,      main)
+:
+dnl	AC_CHECK_LIB(bsd,      main)
 fi
 AC_CHECK_LIB(util,     setproctitle)
 AC_CHECK_LIB(m,        main)
 AC_CHECK_LIB(dl,       main)
-AC_CHECK_LIB(nsl,      main)
+dnl AC_CHECK_LIB(nsl,      main)
 AC_CHECK_LIB(socket,   main)
 AC_CHECK_LIB(ipc,      main)
 AC_CHECK_LIB(IPC,      main)
@@ -630,7 +631,7 @@
 AC_CHECK_LIB(BSD,      main)
 AC_CHECK_LIB(gen,      main)
 AC_CHECK_LIB(PW,       main)
-AC_CHECK_LIB(resolv,   main)
+dnl AC_CHECK_LIB(resolv,   main)
 AC_CHECK_LIB(wsock32,   main)
 AC_SEARCH_LIBS(getopt_long, [getopt gnugetopt])
 # QNX:
@@ -639,7 +640,8 @@
 # BeOS:
 if test "$PORTNAME" = "beos"
 then
-	AC_CHECK_LIB(bind, __inet_ntoa)
+	dnl AC_CHECK_LIB(bind, __inet_ntoa)
+	AC_CHECK_FUNC(inet_ntoa, ,[AC_CHECK_LIB(bind,     __inet_ntoa)]) 
 fi
 # Solaris:
 AC_SEARCH_LIBS(fdatasync, [rt posix4])
