--- postgresql-7.2/src/interfaces/perl5/GNUmakefile~	Sun Feb 10 23:50:38 2002
+++ postgresql-7.2/src/interfaces/perl5/GNUmakefile	Sun Feb 10 23:50:38 2002
@@ -36,9 +36,9 @@
 	$(PERL) $< \
 	  INC='-I$(srcdir) -I$(libpq_srcdir) $(filter -I%, $(CPPFLAGS))' \
 	  LIBS="-L$$abs_libpq_builddir -lpq" \
-	  INSTALLSITEARCH='$$(DESTDIR)$(perl_installsitearch)' \
-	  INSTALLSITELIB='$$(DESTDIR)$(perl_installsitelib)' \
-	  INSTALLMAN3DIR='$$(DESTDIR)$(perl_installman3dir)' \
+	  INSTALLSITEARCH='$(perl_installsitearch)' \
+	  INSTALLSITELIB='$(perl_installsitelib)' \
+	  INSTALLMAN3DIR='$(perl_installman3dir)' \
 	  MAKEFILE="$$abs_builddir/Makefile"
 
 .PHONY: libpq-all
