diff -urN postgresql-7.1.org/configure.in postgresql-7.1/configure.in
--- postgresql-7.1.org/configure.in	Wed Apr 18 14:00:58 2001
+++ postgresql-7.1/configure.in	Wed Apr 18 14:07:04 2001
@@ -17,9 +17,7 @@
 dnl The GNU folks apparently haven't heard that some people don't use
 dnl Texinfo. Use this sorcery to use "docdir" instead of "infodir".
 define([info], [doc])
-define([infodir], [docdir])
 AC_INIT(src/backend/access/common/heaptuple.c)
-undefine([infodir])
 undefine([info])
 
 AC_PREFIX_DEFAULT(/usr/local/pgsql)
@@ -31,6 +29,9 @@
 VERSION='7.1'
 AC_SUBST(VERSION)
 AC_DEFINE_UNQUOTED(PG_VERSION, "$VERSION")
+
+docdir=$infodir
+AC_SUBST(docdir)
 
 unset CDPATH
 
