diff -urN postgresql-7.0.2/src/bin/pgaccess/Makefile postgresql-7.0.2-new/src/bin/pgaccess/Makefile
--- postgresql-7.0.2/src/bin/pgaccess/Makefile	Fri Oct  6 17:55:59 2000
+++ postgresql-7.0.2-new/src/bin/pgaccess/Makefile	Fri Oct  6 17:54:22 2000
@@ -22,7 +22,7 @@
 
 install:
 	$(INSTALL) -d $(DESTDIR)$(LIBDIR)/pgaccess/help
-	$(INSTALL) -d $(DESTDIR)$(LIBDIR)/pgaccess/anguages
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)/pgaccess/languages
 	$(INSTALL) -d $(DESTDIR)$(LIBDIR)/pgaccess/images
 	$(INSTALL) $(INSTL_EXE_OPTS) pgaccess $(DESTDIR)$(BINDIR)/pgaccess
 	$(INSTALL) $(INSTL_EXE_OPTS) main.tcl $(DESTDIR)$(LIBDIR)/pgaccess
