--- postgresql-7.1.3/src/interfaces/python/GNUmakefile~	Mon Nov  5 12:57:54 2001
+++ postgresql-7.1.3/src/interfaces/python/GNUmakefile	Mon Nov  5 13:00:49 2001
@@ -35,6 +35,7 @@
 install: all
 	@echo "Installing Python module"
 	@if ( $(INSTALL_DATA) pg.py $(DESTDIR)$(python_moduledir) && \
+		$(INSTALL_DATA) pgdb.py $(DESTDIR)$(python_moduledir) && \
 	    $(MAKE) -f Makefile LIBDIR=\"$(DESTDIR)/$(prefix)/lib\" install ); then : ; else \
 		echo "*****" ;\
 		echo "* Skipping the installation of the Python interface module for lack"; \
