diff -Nur postgresql-7.0.2.orig/src/Makefile.global.in postgresql-7.0.2/src/Makefile.global.in
--- postgresql-7.0.2.orig/src/Makefile.global.in	Tue May 23 04:12:46 2000
+++ postgresql-7.0.2/src/Makefile.global.in	Wed Jul  5 13:56:19 2000
@@ -62,7 +62,7 @@
 # For convenience, POSTGRESDIR is where BINDIR, and LIBDIR
 # and other target destinations are rooted.  Of course, each of these is
 # changable separately.
-POSTGRESDIR= @prefix@
+POSTGRESDIR= $(DESTDIR)@prefix@
 
 # Where the postgres executables live (changeable by just putting them
 # somewhere else and putting that directory in your shell PATH)
@@ -84,10 +84,10 @@
 IPCRM=@ipcrm@
 
 # Where the man pages (suitable for use with "man") get installed.
-POSTMANDIR= $(POSTGRESDIR)/man
+POSTMANDIR= $(DESTDIR)@mandir@
 
 # Where the formatted documents (e.g., the reference manual) get installed.
-POSTDOCDIR= $(POSTGRESDIR)/doc
+POSTDOCDIR= $(POSTGRESDIR)/share/doc
 
 # Where the header files necessary to build frontend programs get installed.
 HEADERDIR= $(POSTGRESDIR)/include
