diff -urNp -x '*.orig' postgresql-9.3.25.org/src/backend/utils/misc/postgresql.conf.sample postgresql-9.3.25/src/backend/utils/misc/postgresql.conf.sample
--- postgresql-9.3.25.org/src/backend/utils/misc/postgresql.conf.sample	2018-11-05 22:53:28.000000000 +0100
+++ postgresql-9.3.25/src/backend/utils/misc/postgresql.conf.sample	2021-11-16 20:59:56.259079714 +0100
@@ -291,7 +291,7 @@
 
 # - Where to Log -
 
-#log_destination = 'stderr'		# Valid values are combinations of
+log_destination = 'syslog'		# Valid values are combinations of
 					# stderr, csvlog, syslog, and eventlog,
 					# depending on platform.  csvlog
 					# requires logging_collector to be on.
@@ -324,8 +324,8 @@
 					# 0 disables.
 
 # These are relevant when logging to syslog:
-#syslog_facility = 'LOCAL0'
-#syslog_ident = 'postgres'
+syslog_facility = 'LOCAL0'
+syslog_ident = 'postgres'
 
 # This is only relevant when logging to eventlog (win32):
 # (change requires restart)
@@ -385,7 +385,7 @@
 #debug_print_plan = off
 #debug_pretty_print = on
 #log_checkpoints = off
-#log_connections = off
+log_connections = on
 #log_disconnections = off
 #log_duration = off
 #log_error_verbosity = default		# terse, default, or verbose messages
