--- postgresql-9.4.0/configure.in.orig	2015-01-05 23:55:32.000000000 +0100
+++ postgresql-9.4.0/configure.in	2015-01-06 00:03:58.433386714 +0100
@@ -1166,13 +1166,13 @@
         [AC_MSG_ERROR([header file <uuid.h> does not match E2FS UUID library])])],
       [AC_MSG_ERROR([header file <uuid/uuid.h> or <uuid.h> is required for E2FS UUID])])])
 elif test "$with_uuid" = ossp ; then
-  AC_CHECK_HEADERS(ossp/uuid.h,
-    [AC_EGREP_HEADER([uuid_export], ossp/uuid.h, [],
-      [AC_MSG_ERROR([header file <ossp/uuid.h> does not match OSSP UUID library])])],
+  AC_CHECK_HEADERS(ossp-uuid/uuid.h,
+    [AC_EGREP_HEADER([uuid_export], ossp-uuid/uuid.h, [],
+      [AC_MSG_ERROR([header file <ossp-uuid/uuid.h> does not match OSSP UUID library])])],
     [AC_CHECK_HEADERS(uuid.h,
       [AC_EGREP_HEADER([uuid_export], uuid.h, [],
         [AC_MSG_ERROR([header file <uuid.h> does not match OSSP UUID library])])],
-      [AC_MSG_ERROR([header file <ossp/uuid.h> or <uuid.h> is required for OSSP UUID])])])
+      [AC_MSG_ERROR([header file <ossp-uuid/uuid.h> or <uuid.h> is required for OSSP UUID])])])
 fi
 
 if test "$PORTNAME" = "win32" ; then
