diff -urNp -x '*.orig' postgresql-9.6.19.org/src/backend/utils/misc/postgresql.conf.sample postgresql-9.6.19/src/backend/utils/misc/postgresql.conf.sample
--- postgresql-9.6.19.org/src/backend/utils/misc/postgresql.conf.sample	2020-08-10 23:21:12.000000000 +0200
+++ postgresql-9.6.19/src/backend/utils/misc/postgresql.conf.sample	2021-11-24 08:41:24.212264575 +0100
@@ -329,7 +329,7 @@
 
 # - Where to Log -
 
-#log_destination = 'stderr'		# Valid values are combinations of
+log_destination = 'syslog'		# Valid values are combinations of
 					# stderr, csvlog, syslog, and eventlog,
 					# depending on platform.  csvlog
 					# requires logging_collector to be on.
@@ -362,9 +362,9 @@
 					# 0 disables.
 
 # These are relevant when logging to syslog:
-#syslog_facility = 'LOCAL0'
-#syslog_ident = 'postgres'
-#syslog_sequence_numbers = on
+syslog_facility = 'LOCAL0'
+syslog_ident = 'postgres'
+syslog_sequence_numbers = on
 #syslog_split_messages = on
 
 # This is only relevant when logging to eventlog (win32):
