diff -Naur postgresql-7.4.7.orig/configure.in postgresql-7.4.7/configure.in
--- postgresql-7.4.7.orig/configure.in	2005-02-10 12:48:09.683051872 +0100
+++ postgresql-7.4.7/configure.in	2005-02-10 12:47:17.457991288 +0100
@@ -20,9 +20,7 @@
 dnl The GNU folks apparently haven't heard that some people don't use
 dnl Texinfo. Use this sorcery to use "docdir" instead of "infodir".
 m4_define([info], [doc])
-m4_define([infodir], [docdir])
 AC_INIT([PostgreSQL], [7.4.19], [pgsql-bugs@postgresql.org])
-m4_undefine([infodir])
 m4_undefine([info])
 AC_SUBST(docdir)
 
@@ -34,6 +32,8 @@
 AC_SUBST(configure_args, [$ac_configure_args])
 
 AC_DEFINE_UNQUOTED(PG_VERSION, "$PACKAGE_VERSION", [PostgreSQL version])
+docdir=$infodir
+AC_SUBST(docdir)
 
 AC_CANONICAL_HOST
 
diff -Naur postgresql-7.4.7.orig/doc/Makefile postgresql-7.4.7/doc/Makefile
--- postgresql-7.4.7.orig/doc/Makefile	2005-02-10 12:48:09.509078320 +0100
+++ postgresql-7.4.7/doc/Makefile	2005-02-10 12:47:09.336225984 +0100
@@ -25,10 +25,6 @@
 
 .NOTPARALLEL:
 
-ifneq ($(wildcard $(srcdir)/postgres.tar.gz),)
-found_html := yes
-endif
-
 ifneq ($(wildcard $(srcdir)/man.tar.gz),)
 # SCO OpenServer's man system is sufficiently different to not bother.
 ifneq ($(PORTNAME), sco)
