diff -urNp -x '*.orig' postgresql-11.5.org/configure.in postgresql-11.5/configure.in
--- postgresql-11.5.org/configure.in	2021-05-08 22:44:52.028956200 +0200
+++ postgresql-11.5/configure.in	2021-05-08 22:44:54.932351739 +0200
@@ -1401,6 +1401,7 @@ fi
 
 if test "$with_bonjour" = yes ; then
   AC_CHECK_HEADER(dns_sd.h, [], [AC_MSG_ERROR([header file <dns_sd.h> is required for Bonjour])])
+  AC_CHECK_LIB(dns_sd, DNSServiceRegister, [BONJOUR_LIBS=-ldns_sd])
 dnl At some point we might add something like
 dnl AC_SEARCH_LIBS(DNSServiceRegister, dns_sd)
 dnl but right now, what that would mainly accomplish is to encourage
@@ -1408,6 +1409,7 @@ dnl people to try to use the avahi imple
 dnl If you want to use Apple's own Bonjour code on another platform,
 dnl just add -ldns_sd to LIBS manually.
 fi
+AC_SUBST([BONJOUR_LIBS])
 
 # for contrib/uuid-ossp
 if test "$with_uuid" = bsd ; then
diff -urNp -x '*.orig' postgresql-11.5.org/src/Makefile.global.in postgresql-11.5/src/Makefile.global.in
--- postgresql-11.5.org/src/Makefile.global.in	2019-08-05 23:14:59.000000000 +0200
+++ postgresql-11.5/src/Makefile.global.in	2021-05-08 22:44:54.932351739 +0200
@@ -173,6 +173,7 @@ includedir_internal = $(pkgincludedir)/i
 pgxsdir = $(pkglibdir)/pgxs
 bitcodedir = $(pkglibdir)/bitcode
 
+BONJOUR_LIBS            = @BONJOUR_LIBS@
 
 ##########################################################################
 #
diff -urNp -x '*.orig' postgresql-11.5.org/src/backend/Makefile postgresql-11.5/src/backend/Makefile
--- postgresql-11.5.org/src/backend/Makefile	2019-08-05 23:14:59.000000000 +0200
+++ postgresql-11.5/src/backend/Makefile	2021-05-08 22:44:54.932351739 +0200
@@ -60,7 +60,7 @@ ifneq ($(PORTNAME), win32)
 ifneq ($(PORTNAME), aix)
 
 postgres: $(OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) $(LDFLAGS_EX) $(export_dynamic) $(call expand_subsys,$^) $(LIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $(LDFLAGS_EX) $(export_dynamic) $(call expand_subsys,$^) $(LIBS) $(BONJOUR_LIBS) -o $@
 
 endif
 endif
